fix docker build

dingdayu opened the pull request:
docker build:

```
# clone git repository & to dir
git clone https://github.com/furrtek/portapack-havoc.git && cd portapack-havoc

# download git submodule 
git submodule update --init --recursive

# build docker images
docker build -t portapack -f dockerfile .

# build portapack-havoc
docker run -v `pwd`/havocbin:/havocbin portapack

# check bin file
ls havocbin
```

dingdayu committed:
update python to python3

dingdayu committed:
fix portapack-h1-havoc.bin file path

dingdayu committed:
update dockerfile

dingdayu committed:
fix git branch

dingdayu commented:
build in docker:

```
docker run -v `pwd`:/havoc/bin dingdayu/havoc
```

Compile the latest code of the current warehouse to the current directory.

eried commented:
Great! works now in Windows

> docker run -v `pwd`/havocbin:/havocbin portapack

I guess you want to say `havoc/bin` here right? 

In Docker Windows, something like this works:
`docker run -v ${PWD}/build:/havoc/bin portapack`

dingdayu commented:
> 大！现在可以在Windows中使用
> 
> > 泊坞窗运行-v `pwd`/ havocbin：/ havocbin portapack
> 
> 我想你想在`havoc/bin`这里说对吗？
> 
> 在Docker Windows中，类似这样的工作：
> `docker run -v ${PWD}/build:/havoc/bin portapack`

Yeah

eried committed:
GPS Sim

eried committed:
Main menu icon for GPS sim

eried committed:
Moving to transmitters menu

eried committed:
Merge pull request #1 from furrtek/master

Few improvements in ui_transmitter (#332)

eried committed:
Merge pull request #1 from furrtek/master

Few improvements in ui_transmitter (#332)

eried committed:
Create dockerfile-nogit

dockerfile for compiling a local folder that contains the sourcecode, instead of pulling a fresh copy every time.

To prepare to compile Havoc in Windows/Linux/Mac Docker for example, in the path of the sourcecode:
    docker build -t portapack -f dockerfile-nogit .

And then:
    docker run -v ${PWD}:/havoc portapack

The bin file will be on the build directory now.

eried committed:
Update ui_about.hpp

eried committed:
Small alignment fix

eried committed:
Merge pull request #2 from furrtek/master

Update from origin

eried committed:
Merge pull request #3 from jamesshao8/shao-credits

Shao credits

eried committed:
Update README.md

eried committed:
Merge branch 'master' into gps-sim

eried committed:
Merge pull request #4 from jamesshao8/gps-sim

Gps sim

eried committed:
Merge branch 'master' into gps-sim

eried committed:
Merge pull request #5 from jamesshao8/gps-sim

Merge branch 'master' into gps-sim

eried committed:
Merge pull request #6 from furrtek/master

Update

eried committed:
Update README.md

eried committed:
H2 photos

eried committed:
Merge branch 'master' of https://github.com/eried/portapack-havoc

eried committed:
Update README.md

eried committed:
Update README.md

eried committed:
Adding Debug app back

eried committed:
Github pages for the nightly builds

eried committed:
Removing old doc files

eried committed:
Deployment will be managed by app.buddy.works now

eried committed:
Cleaning more old files

eried committed:
Create dockerfile-nogit

eried committed:
Adding CCACHE

eried committed:
Merge branch 'master' into dockerfile

eried committed:
Merge pull request #7 from eried/dockerfile

Dockerfile

eried committed:
Create dockerfile-nogit

eried committed:
Merge branch 'master' into dockerfile-nogit

eried committed:
Merge pull request #8 from eried/dockerfile-nogit

Create dockerfile-nogit

eried committed:
Update world_map.py

eried committed:
GPS sim test files

eried committed:
Merge pull request #10 from eried/gps-test-files

GPS sim test files

eried committed:
Mercator projection

eried committed:
For bigger pictures

eried committed:
Merge pull request #11 from eried/world_map.py

Update world_map.py

eried committed:
Adjust for header

eried committed:
Update ui_geomap.cpp

eried committed:
Update world_map.jpg

Based on openseamaps + openstreetmaps

eried committed:
Small styling change

eried committed:
Merge branch 'world-map-mercator' of https://github.com/eried/portapack-havoc into world-map-mercator

eried committed:
Merge pull request #12 from eried/world-map-mercator

World map mercator

eried committed:
Small cleanup in tools

eried committed:
Replacing the linear "aproximated" way with a proper one (#14)

eried committed:
Removing this examples

They do not work all the time because they have timestamps

eried committed:
Update analog_audio_app.cpp (#16)

eried committed:
Analog audio app m4 guru meditation bug (#18)

* Update analog_audio_app.cpp (#353)

* Update analog_audio_app.cpp

eried committed:
Capture app fix (#19)

* Update analog_audio_app.cpp (#353)

* Update capture_app.cpp

eried committed:
Pocsag improvements (#20)

* Update analog_audio_app.cpp (#353)

* Adding phase field (extracted from @jamesshao8 repo)

nemanjan00 committed:
Do not turn off backlight, while typing

Touch event does not delay automatic backlight off

eried committed:
New pretty font (#23)

* Update analog_audio_app.cpp (#353)

* Update ui_font_fixed_8x16.cpp

eried committed:
Update generate_world_map.bin.py

eried committed:
Merge pull request #22 from nemanjan00/patch-1

Do not turn off backlight, while typing

eried committed:
Update analog_audio_app.cpp (#353) (#25)

eried committed:
Merge pull request #26 from furrtek/master

Update analog_audio_app.cpp (#353)

eried committed:
Adding 2 more steps to the frequency selector (#28)

eried committed:
Removing playdead calls, non-implemented entries in main menu, and small UI tweaks

eried committed:
Long titles corrupt the header

eried committed:
Removing playdead related stuff and small tweaks

eried committed:
Merge branch 'removing-not-implemented-and-play-dead-and-minor-tweaks'

eried committed:
Remote is also not implemented

eried committed:
Merge branch 'removing-not-implemented-and-play-dead-and-minor-tweaks'

eried committed:
Removing old files

eried committed:
Some new icons and colors

eried committed:
This color scheme looks solid, more small tweaks

eried committed:
More small tweaks to the icons

eried committed:
More!

eried committed:
Update index.html

eried committed:
Update CMakeLists.txt

eried committed:
Update issue templates

eried committed:
Update issue templates

eried committed:
Fork rename

eried committed:
Merge branch 'splash-titlebar-rename'

eried committed:
Removing more unnused files

ydixken committed:
fixed the docker build process by installing pip manually

ydixken committed:
fixed the docker build process by installing pip manually

ydixken committed:
fixed the invokation of python for get-pip.py

eried committed:
Merge pull request #32 from ydixken/master

python-pip not available in repositories

eried committed:
Merge pull request #33 from furrtek/master

Sync

eried committed:
For some reason bzip2 is required now

eried committed:
Fixing python in the dockerfiles

For some misterious reason it broke...

eried committed:
Version bump

eried committed:
Progress bar fix

eried committed:
Updating the sdcard folder with the release files

eried committed:
Example sound clips

From www.myinstants.com

eried committed:
Minor UI changes to Soundboard

eried committed:
Version bump

eried committed:
fix OOK transmit - getting stuck after '12E' type 

From https://github.com/z4ziggy/portapack-havoc/commit/395bc486f12b4303cadfc2fa59e9095a409aaaa8

eried committed:
Revert "Adding 2 more steps to the frequency selector (#28)"

This reverts commit ebfcc8189436a13d8e6314eae4cb032417e8014e.

eried committed:
Merge branch 'master' of https://github.com/eried/portapack-mayhem

eried committed:
Python3 compatibility

eried committed:
Tweaking more icons for consistency in style

eried committed:
Fix for https://github.com/eried/portapack-mayhem/issues/35 Capture app hang on exit

eried committed:
Few more icons refreshed

gregoryfenton committed:
Merge pull request #1 from eried/master

Upstream merge 29/05/2020 15:45

jboone committed:
Stop replay before exiting Replay app, to avoid freeze.

eried committed:
Merge pull request #38 from jboone/patch-1

Stop replay before exiting Replay app, to avoid freeze.

eried committed:
Merge branch 'master' of https://github.com/eried/portapack-mayhem

eried committed:
Control alignment on Fileman app

eried committed:
The more arrow looks better with 1 px spacing on some UIs

eried committed:
Version bump

gregoryfenton committed:
Merge pull request #2 from eried/master

Upstream merge

eried committed:
More icon standarization

eried committed:
Update ui_fileman.hpp

Icon for C8 files

gregoryfenton committed:
Merge pull request #3 from eried/master

Upstream sync

eried committed:
New icons

gregoryfenton committed:
Merge pull request #4 from eried/master

New icons

eried committed:
No space after question mark

eried committed:
Friendlier UI in fileman

gregoryfenton committed:
Merge pull request #5 from eried/master

Upstream merge

euquiq committed:
UI menu changes (#41)

* UI menu changes

Changed DEBUG menu style from text / scrollbar into buttons.

* Update ui_debug.cpp

gregoryfenton committed:
Merge pull request #6 from eried/master

UI menu changes (#41)

eried committed:
Update bug_report.md

eried committed:
Updating bug_report to include useful versions information

eried committed:
Removing links from the bug issue template

eried committed:
Update README.md

eried committed:
Update README.md

eried committed:
Update README.md

gregoryfenton committed:
Fix disappearing icons

Icons disappear when they are rolled over using the jog wheel and the colour is set to
ui::color::white()
Fixed by changing the colours of the icons to something other than ui::color::white()

gregoryfenton committed:
Merge pull request #7 from gregoryfenton/gregoryfenton-patch-icon-colours-1

Fix disappearing icons

eried committed:
Update README.md

eried committed:
Update README.md

eried committed:
Link to the internals too from the Readme

gregoryfenton committed:
Add debug menu icons

gregoryfenton committed:
Testing new debug icons

gregoryfenton committed:
Weird characters appeared in hpp file, removed

eried committed:
Merge pull request #47 from gregoryfenton/master

Change icon colours in debug menu

eried committed:
Icons for debug app

euquiq committed:
DEBUG -> PERIPHERALS is now a buttons based menu

eried committed:
Merge pull request #50 from euquiq/H2-eried-branch

DEBUG -> PERIPHERALS is now a buttons based menu

eried committed:
Icon for the peripherals

eried committed:
Update ui_debug.cpp

eried committed:
Update ui_debug.cpp

eried committed:
Update README.md

eried committed:
Adding badges

eried committed:
Update README.md

eried committed:
Adding nightly builds link

johnelder committed:
Scanner update

Renamed scanner squelch to trigger
Added squelch for audio output

johnelder committed:
Scanner update

Renamed scanner squelch to trigger
Added squelch for audio output

johnelder committed:
Merge branch 'master' into scanner-features

eried committed:
Merge pull request #54 from johnelder/scanner-features

Scanner update

eried committed:
New Icon files for Options

eried committed:
Update icon_soundboard.png

eried committed:
Adding missing tools icons

eried committed:
Update ui_scanner.hpp

eried committed:
Update ui_scanner.hpp

eried committed:
Merge branch 'scanner-update'

eried committed:
Dark cyan for generic submenus

eried committed:
Dark cyan for generic submenus

eried committed:
New icons

eried committed:
New icons

eried committed:
Icons and colors changes

eried committed:
Icons and colors changes

johnelder committed:
Scanner features (#55) - Added freq description to scanner display.

johnelder committed:
Scanner features (#55) - Added freq description to scanner display.

eried committed:
Update README.md

eried committed:
Update README.md

eried committed:
Merge pull request #60 from eried/scanner-update

Scanner features (#55) - Added freq description to scanner display.

eried committed:
Update README.md

eried committed:
Update README.md

eried committed:
Update ui_scanner.cpp

eried committed:
Update ui_scanner.cpp

gregoryfenton committed:
Remove clunky text when changing to HackRF mode

eried committed:
Merge branch 'master' of https://github.com/eried/portapack-mayhem

eried committed:
Merge pull request #64 from gregoryfenton/master

Remove clunky text when changing to HackRF mode

eried committed:
Icon for muted speaker

eried committed:
Persistent setting for speaker icon

eried committed:
Mute and unmute audio

eried committed:
Setting for hiding the speaker

eried committed:
Speaker option for the H1

eried committed:
Update README.md

gregoryfenton committed:
Update README.md

eried committed:
Merge pull request #66 from gregoryfenton/patch-1

Update README.md

eried committed:
Update README.md

eried committed:
Update README.md

eried committed:
Update index.html

eried committed:
Update README.md

eried committed:
Debug sd card verbose by @gregoryfenton

eried committed:
Version bump

eried committed:
Update README.md

euquiq committed:
Antenna length Calculator fix

Calculation now rounds up decimals so result is more accurate.

eried committed:
Merge pull request #78 from euquiq/Whip-calc-rounding-values

Antenna length Calculator fix

eried committed:
Update hackrf

eried committed:
Update bug_report.md

euquiq committed:
GPS SIMULATOR EXIT ON PLAY FIX

Now, if you exit the GPS SIMULATOR while transmitting, it will properly stop the radio and won't crash, exiting as expected.

eried committed:
Merge pull request #81 from euquiq/fix-gps-sim-exit-while-transmitting-freeze

GPS SIMULATOR EXIT ON PLAY FIX

eried committed:
Merge branch 'update-hackrf-module'

euquiq committed:
New Antenna length Calculator

It reads the antennas definition from a txt file:

WHIPCALC/ANTENNAS.TXT

Inside the textfile you place each antenna you own with the following sintaxis:

<antenna label> <elements length in mm, separated by a space>

For example:

ANT500 185 315 450 586 724 862

Input the required frequency, adjust the wave type (full / half / quarter, etc.) and the calculator will return the antenna length (metric and imperial) while also calculating how much you need to expand the fitting antennas you got defined on the txt.

It may return up to 8 matching antennas, which is more than enough (normally you will have 2, perhaps 3 telescopic antennas around for your portapack)

If by any chance your antennas txt got more than 8 antennas, and more than 8 matches the length of the freq / wave you want, it will only show the first 8 matching antennas and will warn you at the bottom that there are even more results (hidden).

All calculations now are rounded into the best integer, considering first decimal, so precision is double than the original antenna calculator app.

eried committed:
Merge pull request #85 from euquiq/new-improved-antenna-calculator

New Antenna length Calculator

eried committed:
Comma as separator

eried committed:
Update ANTENNAS.TXT

eried committed:
Update README.md

eried committed:
Update README.md

eried committed:
Update README.md

eried committed:
Title was too long

eried committed:
Show all antennas even if they are off range

eried committed:
Slighty larger steps

eried committed:
Show infinite+ if freq is 0 and small tweaks

eried committed:
Clear with buffer clear

eried committed:
Update ANTENNAS.TXT

eried committed:
Just minor changes

eried committed:
Merge branch 'new-antenna-length-calculator'

eried committed:
Some titles were too long

eried committed:
Version bump

eried committed:
Update README.md

eried committed:
Update README.md

eried committed:
Update README.md

eried committed:
Adding badge with latest release to Readme

euquiq committed:
Fixed bug in datetime

As shown in https://github.com/eried/portapack-mayhem/issues/88 ...

Tiny bug but probably responsible for badly forming datetime in several apps, as it is used in ACARS, POCSAG and ADSB_TX (and of course AIS RX)

eried committed:
Merge pull request #89 from euquiq/fixed-datetime-bug

Fixed bug in datetime

jwetzell committed:
Adjust polar to point and bearing drawing

jwetzell committed:
use screen_rect for center coordinate

eried committed:
Merge pull request #90 from jwetzell/fix-draw-bearing

Fix draw bearing

eried committed:
Update .gitignore

eried committed:
Merge branch 'bearing-fix'

dingdayu committed:
Merge branch 'master' of https://github.com/eried/portapack-mayhem into eried-master

dingdayu committed:
Merge branch 'eried-master'
